app/Main.hs

```diff
@@ -74,14 +77,17 @@ main = do
       runner         = if env |> web |> Web.isTLS then runTLS tlsSettings' else runSettings
       condDebug      = if env |> web |> Web.pretty then identity else logStdoutDev
 
-    runRIO Config {..} do
+    withDBPool baseLogger pgConnectInfo 4 \dbPool -> runRIO Config {..} do
       logDebug . displayShow =<< ask
-
       when (env |> web |> Web.monitor) Monitor.wai
-      liftIO . runner settings
-             . CORS.middleware
-             . condDebug
-             =<< Web.app
+
+      app <- Web.app
+
+      app
+        |> condDebug
+        |> CORS.middleware
+        |> runner settings
+        |> liftIO
```

Comment:
I think this reads better. Applying middleware and whatnot.

![](https://media2.giphy.com/media/KxhIhXaAmjOVy/giphy.gif?cid=36b14facf12cbe88b0460cd5eb4f2add093acde2121015b2&rid=giphy.gif)

Reply:
Right pipes ftw